sdk-springboot/src/main/java/io/dapr/springboot/DaprTopicSubscription.java

```diff
@@ -20,64 +20,78 @@
+   * @param metadata        Metadata for extended subscription functionality.
    */
-  DaprTopicSubscription(String pubsubName, String topic, String route,
-                        Map<String, String> metadata, DaprTopicBulkSubscribe bulkSubscribe) {
-    this(pubsubName, topic, route, null, metadata, bulkSubscribe);
+  DaprTopicSubscription(String pubsubName, String topic, String route, String deadLetterTopic,
+      Map<String, String> metadata, DaprTopicBulkSubscribe bulkSubscribe) {
+    this(pubsubName, topic, route, deadLetterTopic,null, metadata, bulkSubscribe);
   }
 
   /**
    * Create a subscription topic.
-   * @param pubsubName The pubsub name to subscribe to.
-   * @param topic The topic to subscribe to.
-   * @param route Destination route for messages.
-   * @param routes Destination routes with rules for messages.
-   * @param metadata Metadata for extended subscription functionality.
+   *
+   * @param pubsubName      The pubsub name to subscribe to.
+   * @param topic           The topic to subscribe to.
+   * @param route           Destination route for messages.
+   * @param deadLetterTopic Name of topic to forward undeliverable messages.
+   * @param routes          Destination routes with rules for messages.
+   * @param metadata        Metadata for extended subscription functionality.
    */
-  DaprTopicSubscription(String pubsubName, String topic, String route, DaprTopicRoutes routes,
-                        Map<String, String> metadata) {
-    this(pubsubName, topic, route, routes, metadata, null);
+  DaprTopicSubscription(String pubsubName, String topic, String route, String deadLetterTopic,
+      DaprTopicRoutes routes,
+      Map<String, String> metadata) {
+    this(pubsubName, topic, route, deadLetterTopic, routes, metadata, null);
   }
 
   /**
    * Create a subscription topic.
-   * @param pubsubName The pubsub name to subscribe to.
-   * @param topic The topic to subscribe to.
-   * @param route Destination route for messages.
-   * @param routes Destination routes with rules for messages.
-   * @param metadata Metadata for extended subscription functionality.
-   * @param bulkSubscribe Bulk subscribe configuration.
+   *
+   * @param pubsubName      The pubsub name to subscribe to.
+   * @param topic           The topic to subscribe to.
+   * @param route           Destination route for messages.
+   * @param deadLetterTopic Name of topic to forward undeliverable messages.
+   * @param routes          Destination routes with rules for messages.
+   * @param metadata        Metadata for extended subscription functionality.
+   * @param bulkSubscribe   Bulk subscribe configuration.
    */
-  DaprTopicSubscription(String pubsubName, String topic, String route, DaprTopicRoutes routes,
-                        Map<String, String> metadata, DaprTopicBulkSubscribe bulkSubscribe) {
+  DaprTopicSubscription(String pubsubName, String topic, String route, String deadLetterTopic,
```

Comment:
This is the one that contains all params, so this is correct. The others should remain untouched or have new ones added with deadLetterTopic. It is unfortunate that Java requires all permutations of params to be declared as each separate function overload.